examples/Demo.elm

```diff
@@ -57,37 +67,106 @@ type Action
   | ButtonsAction Demo.Buttons.Action
   | TextfieldAction Demo.Textfields.Action
   | SnackbarAction Demo.Snackbar.Action
-  --| TemplateAction Demo.Template.Action
+  | RoutingAction Routing.Action
+  | HopAction ()
+
+
 
+--| TemplateAction Demo.Template.Action
 
-update : Action -> Model -> (Model, Effects.Effects Action)
+
+changeTab : Layout.Action -> Effects Action
+changeTab action =
+  let
+    navTo path =
+      Effects.map HopAction (navigateTo path)
+  in
+    case action of
+      Layout.SwitchTab n ->
+        case n of
+          0 ->
+            navTo "/snackbar"
+
+          1 ->
+            navTo "/textfields"
+
+          2 ->
+            navTo "/buttons"
+
+          3 ->
+            navTo "/grid"
+
+          4 ->
+            navTo "/badges"
+
+          _ ->
+            navTo "/404"
+
+      _ ->
+        Effects.none
+
+
+update : Action -> Model -> ( Model, Effects.Effects Action )
 update action model =
   case Debug.log "Action: " action of
-    LayoutAction    a -> lift  .layout     (\m x->{m|layout    =x}) LayoutAction   Layout.update          a model
-    ButtonsAction   a -> lift  .buttons    (\m x->{m|buttons   =x}) ButtonsAction  Demo.Buttons.update    a model
-    TextfieldAction a -> lift' .textfields (\m x->{m|textfields=x})                Demo.Textfields.update a model
-    SnackbarAction  a -> lift  .snackbar   (\m x->{m|snackbar  =x}) SnackbarAction Demo.Snackbar.update   a model
-    --TemplateAction  a -> lift  .template   (\m x->{m|template  =x}) TemplateAction Demo.Template.update   a model
+    LayoutAction a ->
+      let
+        ( lifted, layoutFx ) =
+          lift .layout (\m x -> { m | layout = x }) LayoutAction Layout.update a model
 
+        routeFx =
+          changeTab a
 
--- VIEW
+        fx =
+          Effects.batch [ layoutFx, routeFx ]
+      in
+        ( lifted, fx )
+
+    ButtonsAction a ->
+      lift .buttons (\m x -> { m | buttons = x }) ButtonsAction Demo.Buttons.update a model
 
+    TextfieldAction a ->
+      lift' .textfields (\m x -> { m | textfields = x }) Demo.Textfields.update a model
 
-type alias Addr = Signal.Address Action
+    SnackbarAction a ->
+      lift .snackbar (\m x -> { m | snackbar = x }) SnackbarAction Demo.Snackbar.update a model
 
+    RoutingAction a ->
+      let
+        ( routing', fx ) =
+          Routing.update a model.routing
+
+        model' =
+          { model | routing = routing' }
+      in
+        ( model'
+        , Effects.map RoutingAction fx
+        )
+
+    HopAction _ ->
+      ( model, Effects.none )
```

Comment:
It looks weird that the HopAction doesn't do anything. Is it because it's really the HopAction in the Router module that counts?